### PR TITLE
update proto to support caveats

### PR DESCRIPTION
**Pull request by vroldanbet**
Updates buf commit to generate the code to support caveats.

Buf CLI, JDK and Gradle versions were also updated.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -21,17 +21,17 @@ jobs:
         distribution: "adopt"
         java-package: "jdk"
         java-version: "${{ matrix.java }}"
-    - uses: "bufbuild/buf-setup-action@v1"
+    - uses: "bufbuild/buf-setup-action@v1.9.0"
```
Comment (jzelinskie) [inline .github/workflows/build.yaml:24]: Don't we want to keep this at v1?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -40,9 +40,9 @@ jobs:
           distribution: "adopt"
           java-package: "jdk"
           java-version: "${{ matrix.java }}"
-      - uses: "bufbuild/buf-setup-action@v1"
+      - uses: "bufbuild/buf-setup-action@v1.9.0"
```
Comment (jzelinskie) [inline .github/workflows/build.yaml:43]: ditto

**Review by jzelinskie — CHANGES_REQUESTED** [review]
I don't think we're supposed to check in the .jar

File: .github/workflows/build.yaml (hunk 1 above)
Comment (vroldanbet) [reply .github/workflows/build.yaml:24]: for whatever reason that's what the README at buf-setup-action showed. It did seem weird they deviated from the established pattern

Comment (vroldanbet) [reply .github/workflows/build.yaml]: @jzelinskie addressed in https://github.com/authzed/authzed-java/pull/32/commits/0c05f4f9b67f881ae5c82f8b0f8d7ea106a3c812

File: .github/workflows/build.yaml (hunk 2 above)
Comment (vroldanbet) [reply .github/workflows/build.yaml]: https://github.com/authzed/authzed-java/pull/32/commits/0c05f4f9b67f881ae5c82f8b0f8d7ea106a3c812

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**